java/flight/flight-core/src/main/java/org/apache/arrow/flight/FlightClient.java

```diff
@@ -192,13 +194,13 @@ public void authenticate(ClientAuthHandler handler, CallOption... options) {
    * @return a CredentialCallOption containing a bearer token if the server emitted one, or
    *     empty if no bearer token was returned. This can be used in subsequent API calls.
    */
-  public Optional<CredentialCallOption> basicHeaderAuthenticate(String username, String password) {
-    final ClientBearerTokenMiddleware.Factory tokenMiddleware = new ClientBearerTokenMiddleware.Factory();
-    middleware.add(tokenMiddleware);
+  public Optional<CredentialCallOption> authenticateBasicToken(String username, String password) {
+    final ClientIncomingAuthHeaderMiddleware.Factory clientAuthMiddleware =
+            new ClientIncomingAuthHeaderMiddleware.Factory(headerHandler);
```

Comment:
Let's make this method use the bearer token header handler. This method is specifically for using the basic auth workflow that attempts to get a bearer token (see the javadoc above).

Someone that wants to use a different outgoing header can go setup a ClientIncomingAuthMiddleware themselves and call handshake themselves. Let's remove the ClientHeaderHandler from the builder.

Reply:
Done.